### PR TITLE
AVRO-1649: More Informative Schema Compatibility Checks

### PR DESCRIPTION
**Pull request by kurtharriger**
Rebased https://github.com/apache/avro/pull/33 onto trunk

https://issues.apache.org/jira/browse/AVRO-1649

### CODE REVIEW

File: lang/java/avro/src/main/java/org/apache/avro/SchemaCompatibility.java [hunk 1]
```diff
@@ -415,6 +421,36 @@ private SchemaCompatibilityType calculateCompatibility(
...
   }
 
+  /**
+   * Identifies the type of a schema compatibility result.
+   */
+  public static enum SchemaCompatibilityResult {
+    COMPATIBLE(READER_WRITER_COMPATIBLE_MESSAGE),
+
+    INCOMPATIBLE_NAME(READER_WRITER_INCOMPATABLE_MESSAGE + ". Schema names must match."),
+    INCOMPATIBLE_SIZE(READER_WRITER_INCOMPATABLE_MESSAGE + ". Fixed schemas are no the same size."),
```
Comment (busbey) [inline lang/java/avro/src/main/java/org/apache/avro/SchemaCompatibility.java]: Should be "are not the same size"

**Comment by busbey** [comment]
Overall approach looks good. I'll need to look at this when I get to a workstation to e.g. check for API impact.

**Comment by kurtharriger** [comment]
bump

**Comment by rdblue** [comment]
@kurtharriger, we're still interested in getting this in. It was just automatically closed by moving the project to git from SVN. You should be able to reopen it or submit a new one to get this change in. Thanks!

**Pull request closed without merging**